src/basic/log.h

```diff
@@ -36,10 +36,11 @@ typedef enum LogTarget{
         _LOG_TARGET_INVALID = -1
 } LogTarget;
 
-#define LOG_REALM_PLUS_LEVEL(realm, level)      \
-        ((realm) << 10 | (level))
-#define LOG_REALM_REMOVE_LEVEL(realm_level)     \
-        ((realm_level >> 10))
+#define LOG_REALM_PLUS_LEVEL(realm, level)  ((realm) << 10 | (level))
+#define LOG_REALM_REMOVE_LEVEL(realm_level) ((realm_level) >> 10)
+#define SYNTHETIC_ERRNO(num)                (1u << 30 | (num))
+#define IS_SYNTHETIC_ERRNO(val)             (1u << 30 & (val))
+#define ERRNO_VALUE(val)                    ((val < 0 ? -val : val) & 255)
 
```

Comment:
i like the idea! maybe prefix this with `LOG_` though? after all this is kinda specific to the logging subsystem, no?

Reply:
Those macro names are pretty long anyway. I think it's OK, if we ever have a conflict, we'll know it, and then we can rename without any issue. It's not exposed externally after all.